### PR TITLE
Fix/my list screen

### PR DESCRIPTION
**Pull request by Jina-n11**

### CODE REVIEW

File: core/bases/src/main/java/com/chocolatecake/util/BindingAdapters.kt [hunk 1]
```diff
@@ -19,13 +22,25 @@ fun <T> RecyclerView.setRecyclerItems(items: List<T>?) {
 
 @BindingAdapter(value = ["app:isVisible"])
 fun View.isVisible(isVisible: Boolean) {
+    Log.i("bb", "isVisible:  $isVisible")
```
Comment (mohammed-akkad) [inline core/bases/src/main/java/com/chocolatecake/util/BindingAdapters.kt]: delete log

File: core/bases/src/main/java/com/chocolatecake/util/BindingAdapters.kt [hunk 2]
```diff
@@ -19,13 +22,25 @@ fun <T> RecyclerView.setRecyclerItems(items: List<T>?) {
...
+@BindingAdapter(value = ["app:isVisibleCardDelete"])
+fun com.google.android.material.card.MaterialCardView.isVisibleCardDelete(isVisible: Boolean) {
+    if (isVisible) {
+        this.visibility = View.VISIBLE
+    } else {
+        this.visibility = View.INVISIBLE
+    }
+}
+
+
```
Comment (mohammed-akkad) [inline core/bases/src/main/java/com/chocolatecake/util/BindingAdapters.kt]: white space

File: core/data/remote/src/main/java/com/chocolatecake/remote/service/MovieService.kt [hunk 3]
```diff
@@ -157,6 +157,7 @@ interface MovieService {
     ): Response<ProfileRemoteDto>
     ///endregion
 
+
```
Comment (mohammed-akkad) [inline core/data/remote/src/main/java/com/chocolatecake/remote/service/MovieService.kt:160]: white space

File: core/data/repository/src/main/java/com/chocolatecake/repository/mappers/cash/myList/LocalListMapper.kt
Comment (mohammed-akkad) [inline core/data/repository/src/main/java/com/chocolatecake/repository/mappers/cash/myList/LocalListMapper.kt:1]: zombie code :)

File: core/domain/usecase/src/main/java/com/chocolatecake/repository/MovieRepository.kt [hunk 4]
```diff
@@ -98,23 +98,13 @@ interface MovieRepository {
...
-    suspend fun getWatchlistByMediaType(mediaType: String): List<MovieEntity>
-    suspend fun addWatchlist(watchlistRequest: WatchlistRequestEntity): Boolean
-
-
     suspend fun addList(name:String): Boolean
-    suspend fun getLists(): List<ListEntity>
-    suspend fun refreshLists()
 
-    suspend fun getMovieList(): List<ListMovieEntity>
+    suspend fun getDetailsList(listId: Int): List<MovieEntity>
```
Comment (mosayed01) [inline core/domain/usecase/src/main/java/com/chocolatecake/repository/MovieRepository.kt:103]:
```suggestion
    suspend fun detailsList(listId: Int): List<MovieEntity>
```

File: feature/home/presentation/home-ui/src/main/res/layout/item_my_list.xml [hunk 5]
```diff
@@ -35,34 +40,84 @@
...
-                    app:layout_constraintEnd_toEndOf="@id/text_view_season_title"
-                    app:layout_constraintStart_toStartOf="@id/text_view_season_title"
-                    app:layout_constraintTop_toBottomOf="@id/text_view_season_title"
-                    tools:text="9 item" />
+                    android:layout_gravity="center_vertical">
+
+<!--                    <ImageButton-->
+<!--                        android:id="@+id/image_button_show_delete"-->
+<!--                        android:layout_width="wrap_content"-->
+<!--                        android:layout_height="wrap_content"-->
```
Comment (mohammed-akkad) [inline feature/home/presentation/home-ui/src/main/res/layout/item_my_list.xml]: dead code :)

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/myList/MyListListener.kt [hunk 6]
```diff
@@ -2,11 +2,12 @@ package com.chocolatecake.viewmodel.myList
 import com.chocolatecake.bases.BaseInteractionListener
 
 interface MyListListener : BaseInteractionListener {
-//    fun onClickFavoriteList(itemId: Int , listType: String = "favorite" , mediaType: String = "tv")
-//    fun onClickWatchlist(itemId: Int , listType: String ="watchlist", mediaType: String = "tv")
+
     fun onClickItem(listId: Int , listType: String = "movie", listName: String = "favorite")
 
     fun onClickNewList()
```
Comment (mohammed-akkad) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/myList/MyListListener.kt:9]: white space

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/myList/MyListViewModel.kt [hunk 7]
```diff
@@ -76,25 +66,76 @@ class MyListViewModel @Inject constructor(
...
+    private fun onCreateUserNewListSuccess(item: Boolean) {
+        sendEvent(MyListUiEvent.ShowSnackBar("New List Was Added Successfully"))
+        getData()
+    }
+
+
+
+    override fun onClickShowDelete() {
+       _state.update { it.copy(isShowDelete = true , error = null  ) }
+        Log.i("bb", "onClickShowDelete:  ${_state.value.isShowDelete}  ")
```
Comment (mohammed-akkad) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/myList/MyListViewModel.kt]: delete log :)

**Review by mosayed01 — APPROVED** [review]

**Review by mohammed-akkad — APPROVED** [review]

File: feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/my_list/MyListFragment.kt [hunk 8]
```diff
@@ -75,6 +75,11 @@ class MyListFragment :
 
     override fun onClickCreate(listName: String) {
         viewModel.onCreateList(listName)
+        createListBottomSheet.dismiss()
+    }
+
+    override fun failCreated(message: String) {
+        showSnackBar(message)
     }
 }
```
Comment (amna97alaa) [inline feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/my_list/MyListFragment.kt:85]: white space

File: feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/my_list_details/MyListDetailsFragment.kt [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+                findNavController().popBackStack()
+            }
+
+            is MyListDetailsUiEvent.ShowSnackBar -> showSnackBar(event.message)
+
+            else -> {}
+        }
+    }
+}
+
```
Comment (amna97alaa) [inline feature/home/presentation/home-ui/src/main/java/com/chocolatecake/ui/my_list_details/MyListDetailsFragment.kt:86]: delete line 86

File: feature/home/presentation/home-ui/src/main/res/layout/fragment_my_list.xml [hunk 10]
```diff
@@ -19,6 +19,7 @@
                 android:elevation="2dp"
```
Comment (amna97alaa) [inline feature/home/presentation/home-ui/src/main/res/layout/fragment_my_list.xml:19]: remove hard code

File: feature/home/presentation/home-ui/src/main/res/layout/fragment_my_list.xml [hunk 11]
```diff
@@ -19,6 +19,7 @@
...
                 android:layout_width="match_parent"
                 android:layout_height="?android:attr/actionBarSize">
+
                 <ImageButton
                     android:id="@+id/backButton"
                     android:layout_width="24dp"
```
Comment (amna97alaa) [inline feature/home/presentation/home-ui/src/main/res/layout/fragment_my_list.xml:25]: remove hard code

File: feature/home/presentation/home-ui/src/main/res/layout/fragment_my_list_details.xml [hunk 12]
```diff
@@ -19,6 +19,7 @@
             android:elevation="2dp"
             android:layout_width="match_parent"
             android:layout_height="?android:attr/actionBarSize">
+
             <ImageButton
                 android:id="@+id/backButton"
                 android:layout_width="24dp"
```
Comment (amna97alaa) [inline feature/home/presentation/home-ui/src/main/res/layout/fragment_my_list_details.xml:25]: remove hard code

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/myList/MyListViewModel.kt [hunk 13]
```diff
@@ -76,25 +64,75 @@ class MyListViewModel @Inject constructor(
...
+        )
+    }
+
+    private fun onCreateUserNewListSuccess(item: Boolean) {
+        sendEvent(MyListUiEvent.ShowSnackBar("New List Was Added Successfully"))
+        getData()
+    }
+
+
+
```
Comment (amna97alaa) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/myList/MyListViewModel.kt:84]: too much white spaces

File: feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/my_list_details/MyListDetailsViewModel.kt [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+        tryToExecute(
+            call = {
+                deleteMovieFromDetailsListUseCase(listId = listId, mediaId = mediaId)
+            },
+            onSuccess =  ::onDeleteMediaSuccess ,
+            onError = ::onError,
+        )
+    }
+
+
```
Comment (amna97alaa) [inline feature/home/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/my_list_details/MyListDetailsViewModel.kt:146]: delete one line of white space :)

**Review by amna97alaa — APPROVED** [review]

**Pull request merged**